addons/account/wizard/account_payment_register.py

```diff
@@ -149,15 +148,53 @@ def _get_batches(self):
             raise UserError(_("You can't open the register payment wizard without at least one receivable/payable line."))
 
         batches = {}
-        for line in lines:
-            batch_key = self._get_line_batch_key(line)
-
-            serialized_key = '-'.join(str(v) for v in batch_key.values())
+        flagged_keys = ()
+
+        def get_serialized_key(batch_key, include_partner_bank_id):
+            values = [str(v) for v in batch_key.values()]
+            if not include_partner_bank_id:
+                values = values[:-1]
+            return '-'.join(values)
+
+        def add_line_to_batches(line, batch_key=None, serialized_key=None):
+            if not batch_key:
+                batch_key = self._get_line_batch_key(line)
+            if not serialized_key:
+                serialized_key = get_serialized_key(batch_key, include_partner_bank_id=True)
             batches.setdefault(serialized_key, {
                 'key_values': batch_key,
                 'lines': self.env['account.move.line'],
+                'total': 0.0,
             })
             batches[serialized_key]['lines'] += line
+            batches[serialized_key]['total'] += line.balance
+            batches[serialized_key]['key_values']['payment_type'] = 'inbound' if batches[serialized_key]['total'] > 0.0 else 'outbound'
+
+        for line in lines:
+            batch_key = self._get_line_batch_key(line)
+            serialized_key = get_serialized_key(batch_key, include_partner_bank_id=False)
```

Comment:
I don't get why you have this parameter `include_partner_bank_id`? Can't you include the partner_bank_id always in the serialized key??

Reply:
This allows to batch lines whit no partner_bank_id with lines with one partner bank partner_bank_id. E.g. on vendor bill with partner_bank_id with a credit note wo partner_bank_id.